### ISSUE TITLE
[xenial] Date / time lost after each reboot

### ISSUE THREAD

**Issue by ernesst** [ubports/ubuntu-touch#437]
After Each reboot, the date goes back to 1970. 

- Device: hammerhead
- Channel: 16.04
- Build: r208

### Expected behavior
Between each reboot the date is kept.

**Comment by Flohack74** [comment]
Oh, not timekeep again! I am afraid we have to repeat the steps we made for some of those devices... For sure its not in 16.04

**Comment by rubencarneiro** [comment]
Ask marius if the timekeep on ubuntu parte is implemented

**Comment by Flohack74** [comment]
@rubencarneiro I found this in the upstart/timekeeper.log:
```
root@ubuntu-phablet:/var/log/upstart# cat timekeeper.log
Cannot create data directory!
Failed to write offset file!
Cannot create data directory!
Failed to write offset file!
Cannot create data directory!
Failed to write offset file!
Cannot create data directory!
Failed to write offset file!
Cannot create data directory!
Failed to write offset file!
```
Does it maybe just need proper permissions?

**Comment by rubencarneiro** [comment]
Timekeep needs permissions to write to /data/timekeep/, and dont have

**Comment by Flohack74** [comment]
@ernesst it is not reproducible for me on r215.

**Comment by ernesst** [comment]
After the update the behavoir was the same. 

After flashing the phone + wipe, it enable the time keeping between reboots. 

Outside of this bug : Would it be possible to know the commit ? I've the same problem on my port - Moto G 2013.

**Comment by Flohack74** [comment]
Yes timezone handlign seems broken... But please lets stay on topic for the original problem, that is, time is not saved between reboots.

**Comment by NeoTheThird** [comment]
Attempted fix by @rubencarneiro: https://github.com/ubports/android_device_fairphone_fp2/pull/10

Maybe @mariogrip can confirm if this was done right so we can avoid shipping a hack. I find the review process on this one insufficient.

**Comment by ernesst** [comment]
It's still the original topic flo, because my sim card is french and i'm leaving in US.
For me the date at each reboot is saved but the time no due to the timezone difference :)

**Comment by ernesst** [comment]
Since R217, i lost the date at the reboot.

**Comment by NeoTheThird** [comment]
Closing this due to [changed issue tracking process](https://github.com/ubports/docs.ubports.com/pull/92). Can someone confirmed this is fixed?

**Comment by NeoTheThird** [comment]
Still an issue on Nexus 5.

**Comment by simoxxx** [comment]
My Nexus4 is able to remember the time and date over reboots.. still not able to save properly the correct time-zone. 
r50. 
Clock core-app still crashing after 3 seconds.

**Comment by zubozrout** [comment]
@simoxxx The clock app worked just fine for me until I set an alarm clock - after this it always loaded but crashed in about 3 seconds just as in your case. I am on MX4 arale.

**Comment by simoxxx** [comment]
I never had the possibility to set the alarm on xenial.. so not the same..

**Comment by zubozrout** [comment]
@simoxxx Well, for some reason the list of alarm clocks was empty on my installation ... none defined. Maybe it has something to do with that. I am not able to look at it now but it may be worth trying to locate the config file in which the alarm clocks are defined and deleting it.

**Comment by simoxxx** [comment]
@zubozrout  Thanks ...I'll try

**Comment by Flohack74** [comment]
Please keep this bug for the timezone issues only, and discuss all app-related stuff in the tracker of clock-app.

**Comment by UniversalSuperBox** [comment]
I've made a small progress on this.

To start, the `timekeeper` service specifies `start on android` in its Upstart job config. I searched as far and wide as I could but simply could not find a job that emitted this signal in 16.04 (it would have the `emits android` line in its config to make Upstart aware of its capability). So, `timekeeper` and a few other jobs never started even though the container was running.

It appears that this signal was changed to `android-container` at some point and other jobs were never updated. By replacing all `start on android` configurations with `start on android-container` configs, they all started correctly! That includes timekeeper!

Unfortunately, that doesn't completely fix the problem. Even though timekeep is now running (and stopping!), it's still not saving the time offset. It doesn't log anything about the failure and exits with status 0.

**Comment by UniversalSuperBox** [comment]
Closing for QA

**Comment by NeoTheThird** [comment]
Still an issue on the FP2 in r219.

**Comment by UniversalSuperBox** [comment]
Seems like the xenial branch isn't used on the timekeeper repository. I've merged the changes into master and they seem to have carried into the repository. These *should* be built into the next image.

I did have to factory reset my phone in order for it to start working, but then again that might have been because I installed the package manually. Closing for QA, let me know if I broke it. :)
